modules/sharedaddy/sharing-sources.php

```diff
@@ -738,9 +738,19 @@ public function display_footer() {
 			if ( ! $locale ) {
 				$locale = 'en_US';
 			}
-			?>
-			<div id="fb-root"></div>
-			<script>(function(d, s, id) { var js, fjs = d.getElementsByTagName(s)[0]; if (d.getElementById(id)) return; js = d.createElement(s); js.id = id; js.src = '//connect.facebook.net/<?php echo $locale; ?>/sdk.js#xfbml=1&appId=249643311490&version=v2.3'; fjs.parentNode.insertBefore(js, fjs); }(document, 'script', 'facebook-jssdk'));</script>
+			/**
+			 * Filter the App ID used in the official Facebook Share button.
+			 *
+			 * @since 3.8.0
+			 *
+			 * @param int $fb_app_id Facebook App ID. Default to empty.
+			 */
+			$fb_app_id = absint( apply_filters( 'jetpack_sharing_facebook_app_id', '249643311490' ) );
```

Comment:
Is calling `absint` going to be a problem with large app IDs? For example, my own example app ID is "1485813205022947", which is larger than `PHP_INT_MAX` (`2147483647`) on 32-bit devices. Maybe instead, we verify numeric value using [`is_numeric`](http://php.net/manual/en/function.is-numeric.php), which can support any number of digits.